senders/integration_test.go

```diff
@@ -46,3 +48,53 @@ func TestTLSEndToEnd(t *testing.T) {
 	assert.Equal(t, 1, len(testServer.MetricLines))
 	assert.Equal(t, "\"my-metric\" 20 source=\"localhost\"", testServer.MetricLines[0])
 }
+
+func TestEndToEndWithInternalMetrics(t *testing.T) {
+	waitTill1stInternalMetricsCollection := time.Millisecond * 1500
+	internalMetricsInterval := time.Millisecond * 900
+
+	testServer := startTestServer()
+	defer testServer.Close()
+
+	sender, err := NewSender(testServer.URL, InternalMetricsInterval(internalMetricsInterval))
+	require.NoError(t, err)
+	require.NoError(t, sender.SendMetric("my metric", 20, 0, "localhost", nil))
+	time.Sleep(waitTill1stInternalMetricsCollection)
+	require.NoError(t, sender.Flush())
+	metricLines := testServer.MetricLines
+
+	assert.Equal(t, true, HelperInternalMetricFound(metricLines, "points.valid"))
+	assert.Equal(t, 12, len(metricLines))
+	assert.Equal(t, "\"my-metric\" 20 source=\"localhost\"", metricLines[0])
+	assert.Equal(t, "/report?f=wavefront", testServer.LastRequestURL)
+}
+
+func TestEndToEndWithInternalMetricsDisabled(t *testing.T) {
+	waitTill1stInternalMetricsCollection := time.Millisecond * 1500
+	internalMetricsInterval := time.Millisecond * 900
+
+	testServer := startTestServer()
+	defer testServer.Close()
+
+	sender, err := NewSender(testServer.URL, InternalMetricsInterval(internalMetricsInterval), InternalMetricsEnabled(false))
+	require.NoError(t, err)
+	require.NoError(t, sender.SendMetric("my metric", 20, 0, "localhost", nil))
+	time.Sleep(waitTill1stInternalMetricsCollection)
+	require.NoError(t, sender.Flush())
+	metricLines := testServer.MetricLines
+
+	assert.Equal(t, false, HelperInternalMetricFound(metricLines, "points.valid"))
+	assert.Equal(t, 1, len(metricLines))
+	assert.Equal(t, "\"my-metric\" 20 source=\"localhost\"", metricLines[0])
+	assert.Equal(t, "/report?f=wavefront", testServer.LastRequestURL)
+}
+
+func HelperInternalMetricFound(metricLines []string, internalMetricName string) bool {
```

Comment:
no need to export this, and you might be able to move it to the test server, so you could call it like...

`testServer.HasReceivedMetric("foo.biz.baz")`

Reply:
I think this change makes the code easier to read, thanks!